gcloud-java-contrib/gcloud-java-nio/pom.xml

```diff
@@ -92,26 +92,44 @@
           <compilerArgument>-Xlint:unchecked</compilerArgument>
         </configuration>
       </plugin>
-      <plugin>
-        <artifactId>maven-jar-plugin</artifactId>
-        <version>2.6</version>
-        <configuration>
-          <archive>
-            <addMavenDescriptor>true</addMavenDescriptor>
-            <index>true</index>
-            <manifest>
-              <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
-              <addDefaultSpecificationEntries>true</addDefaultSpecificationEntries>
-            </manifest>
-            <manifestEntries>
-              <artifactId>${project.artifactId}</artifactId>
-              <groupId>${project.groupId}</groupId>
-              <version>${project.version}</version>
-              <buildNumber>${buildNumber}</buildNumber>
-            </manifestEntries>
-          </archive>
-        </configuration>
-      </plugin>
+        <!-- shade the jar so we can demo adding the NIO jar to add functionality. -->
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-shade-plugin</artifactId>
+            <version>2.4.3</version>
+            <configuration>
+                <shadedArtifactAttached>true</shadedArtifactAttached>
+                <relocations>
+                    <relocation>
+                        <pattern>com</pattern>
+                        <shadedPattern>shaded.gcloud-nio.com</shadedPattern>
+                        <excludes>
+                            <exclude>com.google.gcloud.*</exclude>
```

Comment:
I didn't try that. I couldn't find the documentation for exclude so I went with what I knew worked.